### PR TITLE
Session tracking delivery logic

### PR DESCRIPTION
**Pull request by kinbiko**
## Goal

In order to add session tracking to this notifier we need to be able to send the session payloads to session.bugsnag.com. This change establishes that logic along with the necessary configuration updates.

## Design

There now exists a non-exported `deliverSessions()` function that is intended to be called at regular intervals which will publish the provided sessions. It may be reasonable to make this function into a method in the future for additional testability, but it is currently left as simple as possible, i.e. no receiver.

The configuration parameter `Endpoint` has now been deprecated in favour of `Endpoints` - an analogous change to what has been made to other platforms.

### Added

- `Configuration.Endpoints` which relies on the `Endpoints` struct where you can specify the `Notify` and `Sessions` endpoints for on-prem users.
- `deliverSessions([]session, Configuration)` function that publishes a single session payload containing the count and when the sessions started happening.
- `Configuration.AutoCaptureSessions` (default `true`) which allows automatic session tracking to be disabled.
- BUGSNAG_SESSIONS_ENDPOINT environment variable.

### Deprecated

- `bugsnag.Configuration.Endpoint` will now log a warning if it is being used.

### Changed

- The delivery mechanism of regular payloads has been changed from a `http.Client.Post` to a `http.Client.Do` which allows for more configurability. In particular, it now attaches the `Bugsnag-` prefixed headers:
  - `Bugsnag-Api-Key`
  - `Bugsnag-Payload-Version`
  - `Bugsnag-Sent-At`
- All usages of `Endpoint` have now been switched to `Endpoints`
- The environment variable `BUGSNAG_ENDPOINT` has been deprecated in favour of `BUGSNAG_NOTIFY_ENDPOINT`

## Discussion

What exactly are the `BUGSNAG_FOO_ENDPOINT` environment variables for? They don't appear to be used outside of tests.

## Tests

Added automated tests for:

- The new headers
- The new session payloads and ensuring they get formatted as JSON appropriately when hitting a server.
- The deprecation message when `Endpoint` is being used.

Struggling to run integration tests - even on `master`.

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [ ] Final review

Leaving out the change log as this isn't going to `master`, and thus has no associated version as of yet.

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [x] Usage friction - is the proposed change in usage cumbersome or complicated?
- [x] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [x] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [x] Thoroughness of added tests and any missing edge cases
- [x] Idiomatic use of the language

### CODE REVIEW

File: configuration.go [hunk 1]
```diff
@@ -109,6 +133,16 @@ func (config *Configuration) update(other *Configuration) *Configuration {
 	if other.Logger != nil {
 		config.Logger = other.Logger
 	}
+	if other.Endpoint != "" {
+		config.Logger.Printf("WARNING: the Bugsnag configuration parameter 'Endpoint' is deprecated in favor of 'Endpoints'")
+		config.Endpoint = other.Endpoint
```
Comment (kinbiko) [inline configuration.go]: If this is set then `config.Endpoints.Sessions` should be  unset, and disable session tracking.

File: session.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"runtime"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
```
Comment (kinbiko) [inline session.go]: Fix import order to match `go fmt`
